src/plugins/kibana_usage_collection/server/collectors/core/core_usage_collector.ts

```diff
@@ -0,0 +1,145 @@
+            certificateAuthoritiesConfigured: { type: 'boolean' },
+            certificateConfigured: { type: 'boolean' },
+            keyConfigured: { type: 'boolean' },
+            keystoreConfigured: { type: 'boolean' },
+            truststoreConfigured: { type: 'boolean' },
+            alwaysPresentCertificate: { type: 'boolean' },
+          },
+          apiVersion: { type: 'keyword' },
+          healthCheckDelayMs: { type: 'long' },
+        },
+
+        http: {
+          basePathConfigured: { type: 'boolean' },
+          maxPayloadInBytes: { type: 'long' },
+          rewriteBasePath: { type: 'boolean' },
+          keepaliveTimeout: { type: 'long' },
+          socketTimeout: { type: 'long' },
+          compression: {
+            enabled: { type: 'boolean' },
+            referrerWhitelistConfigured: { type: 'boolean' },
+          },
+          xsrf: {
+            disableProtection: { type: 'boolean' },
+            whitelistConfigured: { type: 'boolean' },
+          },
+          requestId: {
+            allowFromAnyIp: { type: 'boolean' },
+            ipAllowlistConfigured: { type: 'boolean' },
+          },
+          ssl: {
+            certificateAuthoritiesConfigured: { type: 'boolean' },
+            certificateConfigured: { type: 'boolean' },
+            cipherSuites: { type: 'array', items: { type: 'keyword' } },
+            keyConfigured: { type: 'boolean' },
+            keystoreConfigured: { type: 'boolean' },
+            truststoreConfigured: { type: 'boolean' },
+            redirectHttpFromPortConfigured: { type: 'boolean' },
+            supportedProtocols: { type: 'array', items: { type: 'keyword' } },
+            clientAuthentication: { type: 'keyword' },
+          },
+        },
+
+        logging: {
+          appendersTypesUsed: { type: 'array', items: { type: 'keyword' } },
+          loggersConfiguredCount: { type: 'long' },
+        },
+
+        savedObjects: {
+          maxImportPayloadBytes: { type: 'long' },
+          maxImportExportSizeBytes: { type: 'long' },
+        },
+      },
+      environment: {
+        memory: {
+          heapSizeLimit: { type: 'long' },
+          heapTotalBytes: { type: 'long' },
+          heapUsedBytes: { type: 'long' },
+        },
+        os: {
+          distro: { type: 'keyword' },
+          distroRelease: { type: 'keyword' },
+          platform: { type: 'keyword' },
+          platformRelease: { type: 'keyword' },
+        },
+      },
+    },
+    fetch() {
+      return getCoreUsageDataService().getCoreUsageData();
+    },
+
+    /*
+     * Format the response data into a model for internal upload
+     * 1. Make this data part of the "kibana_stats" type
+     * 2. Organize the payload in the usage namespace of the data payload (usage.index, etc)
+     */
+    formatForBulkUpload: (result) => {
+      return {
+        type: KIBANA_STATS_TYPE,
+        payload: {
+          core: result,
+        },
+      };
+    },
```

Comment:
I think this not needed in this case (it's only used for a deprecated piece of logic in `monitoring` (mental note to update the TS documentation to deprecate this method).